pkg/actions/karpenter/karpenter.go

```diff
@@ -1,40 +1,65 @@
 package karpenter
 
 import (
+	"fmt"
 	"time"
 
-	"github.com/weaveworks/eksctl/pkg/karpenter"
-	"github.com/weaveworks/eksctl/pkg/utils/waiters"
-
 	"github.com/aws/aws-sdk-go/aws/request"
-	"k8s.io/client-go/kubernetes"
+	"github.com/kris-nova/logger"
 
 	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
 	"github.com/weaveworks/eksctl/pkg/cfn/manager"
 	"github.com/weaveworks/eksctl/pkg/eks"
+	"github.com/weaveworks/eksctl/pkg/karpenter"
+	"github.com/weaveworks/eksctl/pkg/karpenter/providers/helm"
+	"github.com/weaveworks/eksctl/pkg/utils/tasks"
+	"github.com/weaveworks/eksctl/pkg/utils/waiters"
 )
 
+// Installer contains all necessary dependencies for the Karpenter Install tasks and others.
 type Installer struct {
-	stackManager       manager.StackManager
-	ctl                *eks.ClusterProvider
-	cfg                *api.ClusterConfig
-	clientSet          kubernetes.Interface
-	wait               WaitFunc
-	kubeProvider       eks.KubeProvider
-	karpenterInstaller karpenter.InstallKarpenter
+	StackManager       manager.StackManager
+	CTL                *eks.ClusterProvider
+	Config             *api.ClusterConfig
+	Wait               WaitFunc
+	KarpenterInstaller karpenter.InstallKarpenter
```

Comment:
The whole naming around these is confusing. It could be better. Feel free to spew ideas at me. :D

Reply:
how about `Install karpenter.InstallKarpenter` ? just suggesting tho, naming is hard :D

Reply:
do we need an `installer` struct here? If it only has the 1 func to create the task can just be a public func with a big `InstallOptions` struct for args

Reply:
I hate that option :(

Reply:
Let me elaborate. It's a lot harder to work around a public function and makes a for a rigid structure which can be a pain than to move away from. Especially since then I will have to have a local function as well actually which holds the correct one. And just adds complexity to things. I would rather have a slight naming problem then a bigger problem with a public function. :)